engine/src/flutter/lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -581,7 +585,36 @@ abstract class SemanticRole {
   ///
   /// This boolean decides whether to set the `pointer-events` CSS property to
   /// `all` or to `none` on the semantics [element].
+  ///
+  /// The decision follows a multi-tier policy:
+  ///
+  /// **Tier 1: Explicit Behaviors** - Behaviors attached to this role can
```

Comment:
Declaration should probably come first before this. Also I think you meant `Implicit behaviors`? since the behavior is inferred from semantics behavior/role

Reply:
also ideally we should infer the `SemanticsHitTestBehavior` from the framework side if not provided. That way the platform just need to deal with SemanticsHitTestBehavior